components/fastly_backends/main.tf

```diff
@@ -15,48 +21,76 @@ locals {
   }
 }
 
-resource "fastly_service_v1" "backends-qa" {
-  name          = "Terraform: Backends (QA)"
+resource "fastly_service_v1" "backends" {
+  name          = "Terraform: Backends"
   force_destroy = true
 
-  domain {
-    name = var.northstar_domain
-  }
+  # Configure backend, domain, and condition for each app:
+  dynamic "domain" {
+    for_each = var.backends
 
-  domain {
-    name = var.rogue_domain
+    content {
+      name = domain.value.domain
+    }
   }
 
-  condition {
-    type      = "REQUEST"
-    name      = "backend-northstar-qa"
-    statement = "req.http.host == \"${var.northstar_domain}\""
+  dynamic "condition" {
+    for_each = var.backends
+
+    content {
+      type      = "RESPONSE"
+      name      = "response-${condition.value.name}"
+      statement = "req.http.host == \"${condition.value.domain}\""
+    }
   }
 
-  condition {
-    type      = "RESPONSE"
-    name      = "response-northstar-qa"
-    statement = "req.http.host == \"${var.northstar_domain}\""
+  dynamic "condition" {
+    for_each = var.backends
+
+    content {
+      type      = "REQUEST"
+      name      = "backend-${condition.value.name}"
+      statement = "req.http.host == \"${condition.value.domain}\""
+    }
   }
 
-  condition {
-    type      = "REQUEST"
-    name      = "backend-rogue-qa"
-    statement = "req.http.host == \"${var.rogue_domain}\""
+  dynamic "backend" {
+    for_each = var.backends
+
+    content {
+      address           = backend.value.backend
+      name              = backend.value.name
+      request_condition = "backend-${backend.value.name}" # defined above.
+      port              = 443
+
+      # We use shielding to increase chances that we get cache hits. We don't rely
+      # on Fastly for load balancing (since that is handled by Heroku's router).
+      shield           = "bwi-va-us"
+      auto_loadbalance = false
+    }
```

Comment:
It's awkward to read with how the diff is broken up, but this basically replaces individual `domain { … }`, `condition { … }` and `backend { … }` blocks for each application with a "dynamic" block that builds all of these based on the provided `var.backends` variable.